### ISSUE TITLE
`.repeated().repeated()` hangs silently

### ISSUE THREAD

**Issue by Seldom-SE** [zesterer/chumsky#393]
In `1.0.0-alpha.3`, I've found that `.repeated().repeated()` hangs silently during parsing. Here's a simplified version of the code where I discovered it:

```rust
use chumsky::prelude::*;

#[derive(Debug)]
pub struct Values;

pub fn parser<'a>() -> impl Parser<'a, &'a str, Values, extra::Err<Simple<'a, char>>> {
    let comments = just("//").ignored().repeated();

    any()
        .ignored()
        .padded_by(comments.repeated())
        .repeated()
        .collect::<Vec<_>>()
        .map(|_| Values)
}

fn main() {
    let src = "Would you rather have a CurseForge or a hex editor?";

    println!("{src}");

    println!("Creating parser...");
    let parser = parser().then_ignore(end());
    println!("Parsing file...");
    let result = parser.parse(src);
    println!("Done!");

    if let Some(value) = result.output() {
        println!("{value:?}");
    }
    for error in result.errors() {
        println!("{error}");
    }
}
```

It prints `Creating parser...` and `Parsing file...`, but then gets stuck.

Thanks for the awesome crate!

**Comment by zesterer** [comment]
This is correct behaviour (although I can understand that it feels a little surprising).

`repeated` is permitted to accept a pattern any number of times, including none at all.

This means that a single `x.repeated()` can actually never fail: even if it doesn't see any of `x`, it can always fall back to accepting nothing (this is useful: it can do things like parsing lists with no items).

If you add another `.repeated()` on the end, things get a little funky. Now we're asking for a potentially infinite number of items from a parser that can accept nothing at all: so the second repeated keeps spinning around successfully pulling nothing out of the inner repeating parser.

TL;DR: don't do this, it's pretty much always incorrect. In fact, we're working on implementing a linter (see #280) that will catch cases like this before the parser even runs. Additionally, #368 implements a runtime check that will catch cases like this when compiling with debug mode (but this has not been released in an alpha release yet).

**Comment by Seldom-SE** [comment]
A linter/runtime check resolves this perfectly!
